### PR TITLE
Update DG implementation + Effort

### LINKED ISSUE #219: Update DG Effort + Implementation (Switch Command)
Include >2 images

### PR DESCRIPTION
**Pull request by chrystalquek**
Fix #219

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -195,38 +195,53 @@ The following sequence diagram shows how a list is autosorted upon an addition o
...
-Productiv can be in different modes: dashboard, deliverable, meeting and contact mode. 
-Based on the current mode, the user input is passed to the relevant `LogicManager`. 
-Following that, the `LogicManager` will parse the user input and produce the relevant results.
-The current mode is represented by a `ModeEnum` and stored in `MainWindow`.
+The user input is handled and retrieved by the `MainWindow` and then passed to `LogicModeManager`.
+`LogicModeManager` will call `ModeParser`, which will parse the input and create a `SwitchCommandParser`. `SwitchCommandParser` will return a `SwitchCommand`.
+`LogicModeManager` will then execute `SwitchCommand` which returns a `CommandResult` containing the mode that the app should switch to. 
+Then, `MainWindow` gets the new mode to switch to from `CommandResult`.
+Based on the mode, `MainWindow` will update its own attribute `mode`.
+`MainWindow` will then update the UI to show only related information to the new mode.
```
Comment (gabztcr) [inline docs/DeveloperGuide.md:211]: Can paraphrase to "...update the UI to only show information related to the new mode."

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -195,38 +195,53 @@ The following sequence diagram shows how a list is autosorted upon an addition o
...
 
+##### Aspect: How Switch commands should be implemented
+
+* **Alternative 1 (current choice):** Shortened user commands: `switch` `db`, `dv`, `m` or `c`.
+  * Pros: More convenient and faster to type shorter user commands.
+  * Cons: More difficult for users to remember short forms.
+
+* **Alternative 2 (original implementation):** Longer user commands: `switch` `dashboard`, `deliverable`, `meeting` or `contact`.
+  * Pros: Clearer as commands correspond to the naming of tabs on the navigation bar.
+  * Cons: Take longer to type longer user commands.
```
Comment (gabztcr) [inline docs/DeveloperGuide.md:232]: May want to paraphrase to "Takes longer to type the user commands."

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -788,37 +802,70 @@ testers are expected to do more *exploratory* testing.
 
 1. Deliverables, meetings and contacts are saved automatically to ./data/.
 
-   On normal usage, 3 JSON files are created:
-      * `contactbook.json`
-      * `meetingbook.json`
-      * `deliverablebook.json`
-
+   On normal usage, 3 JSON files are created / saved - `contactbook.json`, `meetingbook.json` and `deliverablebook.json`.
```
Comment (gabztcr) [inline docs/DeveloperGuide.md:805]: May want to reorder to deliverable > meeting > contact files

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -788,37 +802,70 @@ testers are expected to do more *exploratory* testing.
...
 
-   1. Test case: Starting and close the app immediately.<br>
-      Expected: No JSON files created.
+   1. Test case: Start and close the app immediately.<br>
+      Expected: The 3 JSON files are not created.
 
-   1. Test case: Start the app. Switch to deliverable mode. Add a deliverable. Close the app.
-      Expected: Only `deliverablebook.json` created.
+   1. Test case: Start the app. Switch to deliverable mode. Add a deliverable. Close the app.<br>
+      Expected: Of the 3 JSON files, only `deliverablebook.json` created.
```
Comment (gabztcr) [inline docs/DeveloperGuide.md:816]: Can edit to "only `deliverablebook.json` is created."

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -788,37 +802,70 @@ testers are expected to do more *exploratory* testing.
...
 1. Dealing with missing/corrupted data files
 
-   1. Test case: Delete `deliverablebook.json` file and start the jar file again<br>
-      Expected: Data file should re-initialise a list of sample deliverables
+   1. Test case: Delete `deliverablebook.json` file. Start the app. Switch to deliverable mode. Enter `list`. Close the app.<br>
+      Expected: `deliverablebook.json` should re-initialise a list of sample deliverables.
 
-   1. Test case: Corrupt `deliverablebook.json` under ./data/. The easiest way is to add - to a saved deliverable's milestone.
-      Expected: Similar to previous.
+   1. Test case: Corrupt `deliverablebook.json` under ./data/. The easiest way is to add - to a saved deliverable's milestone.<br>
```
Comment (gabztcr) [inline docs/DeveloperGuide.md:823]: Do you mean: "The easiest way is to add a dash (`-`) to a saved deliverable's milestone."?

Also, not very sure if "easiest way" may imply other ways. May want to skip to just "Add a dash (`-`) to a saved..."

**Review by gabztcr — APPROVED** [review]
LGTM with some comments. 
(Did not evaluate your diagrams. We can review again as a whole with all the other diagrams in DG)

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (chrystalquek) [reply docs/DeveloperGuide.md:816]: Concerned that it might be misinterpreted as preferences.json and a bunch of other .json files will be created

**Pull request merged**